# HG changeset patch
# User Stefan Mahr <dac922@gmx.de>
# Date 1395700870 -3600
#      Mon Mar 24 23:41:10 2014 +0100
# Node ID 75467145096fc7b65a9f81207053990d0a27f0ad
# Parent  d0d9f6daa4b6fd61ec4bd26877a0168a6b9c8485
Fix segfault if subsasgn in @class returns no output (bug #41945)

* libinterp/octave-value/ov-usr-fcn.cc (octave_user_function::subsasgn_optimization_ok):
  Check if param_list or ret_list is valid.

diff --git a/libinterp/octave-value/ov-usr-fcn.cc b/libinterp/octave-value/ov-usr-fcn.cc
--- a/libinterp/octave-value/ov-usr-fcn.cc
+++ b/libinterp/octave-value/ov-usr-fcn.cc
@@ -685,16 +685,17 @@ octave_user_function::special_expr (void
   return stmt->expression ();
 }
 
 bool
 octave_user_function::subsasgn_optimization_ok (void)
 {
   bool retval = false;
   if (Voptimize_subsasgn_calls
+      && param_list && ret_list
       && param_list->length () > 0 && ! param_list->varargs_only ()
       && ret_list->length () == 1 && ! ret_list->takes_varargs ())
     {
       tree_identifier *par1 = param_list->front ()->ident ();
       tree_identifier *ret1 = ret_list->front ()->ident ();
       retval = par1->name () == ret1->name ();
     }
 
